src/app/public/modules/modal/modal.service.ts

```diff
@@ -32,15 +34,37 @@ export class SkyModalService {
     private dynamicComponentService?: SkyDynamicComponentService
   ) { }
 
-  // Open Overloads
+  /**
+   * @private
+   * Removes the modal host from the DOM.
+   */
+  public dispose(): void {
+    if (SkyModalService.host) {
+      this.dynamicComponentService.removeComponent(SkyModalService.host);
+      SkyModalService.host = undefined;
+    }
+  }
+
+  /**
+   * Opens a modal using the specified component.
+   * @param component Determines the component to render.
+   * Since you generate the component dynamically instead of with HTML selectors,
```

Comment:
I wonder if we should change this sentence to "Since the component generates dynamically instead of with HTML selectors, consumers must..."